cmd/zpool/zpool_main.c

```diff
@@ -676,6 +676,16 @@ zpool_do_add(int argc, char **argv)
 		return (1);
 	}
 
+	if (!nvlist_exists(props, ZPOOL_CONFIG_ASHIFT)) {
+		char ashift[ZPOOL_MAXPROPLEN];
+		if (zpool_get_prop(zhp, ZPOOL_PROP_ASHIFT, ashift,
```

Comment:
I hadn't thought of that, but now that i'm implementing it i see we would still need to make the `int ashift` a string because `add_prop_list()` needs a `char *propval`. The good thing is that we don't hardcode the default value to "0" here.